### PR TITLE
Update CRDs/velero deps to main for 1.12 work

### PR DESCRIPTION
**Pull request by sseago**

### CODE REVIEW

**Comment by sseago** [comment]
Note: don't merge until we get CI enabled for this branch.

File: controllers/bucket_controller.go [hunk 1]
```diff
@@ -137,7 +137,7 @@ func (b BucketReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl
 	bucket.Status.LastSynced = &v1.Time{Time: time.Now()}
 	bucket.Status.Name = bucket.Spec.Name
 
-	b.Client.Status().Update(ctx, &bucket, &client.SubResourceUpdateOptions{})
+	b.Client.Status().Update(ctx, &bucket, &client.UpdateOptions{})
```
[the remark this answers is not in the document]
Comment (sseago) [reply controllers/bucket_controller.go:140]: Here's what happens without the change:
```
controllers/bucket_controller.go:140:49: undefined: client.SubResourceUpdateOptions
```
I'm basically reverting the change made to this line when we went from k8s/api 0.25 to 0.26 since this change takes us back to 0.25.

We went to 0.26 in oadp-operator to accomodate volsync 7.1 needs, but upstream main code is actually incompatible with this, and with oadp datamover/volsync gone, we were able to move k8s/api back to 0.25 to resolve the errors with imported velero code, which required this line to change back as well.

[the remark this answers is not in the document]
Comment (shawn-hurley) [reply controllers/bucket_controller.go:140]: IIUC, we are saying that we are moving the code back to tracking upstream code and that this change was a carried patch.

[the remark this answers is not in the document]
Comment (sseago) [reply controllers/bucket_controller.go:140]: It wasn't a carried patch, since this is code in oadp-operator, not velero. But this particular change reverts the change we made to this line when we upgraded k8s/api (to meet volsync requirements). The issue is that we're importing velero code from main that is incompatible with the newer 0.26 k8s/api code, but the code in question was not in 1.11, so it didn't come up as a problem when we upgraded the lib for volsync.

In other words, this code is changing not to track upstream velero code but in response to downgrading another lib. Actually, in this case, controller-runtime, but it went along with the k8s lib changes. Those other lib downgrades were required because the newly-pulled-in velero code is incompatible with the version we moved to for volsync.

[the remark this answers is not in the document]
Comment (sseago) [reply controllers/bucket_controller.go:140]: Or to state it slightly more abstractly, we had to upgrade controller-runtime and k8s libs to a newer-than-velero-uses version to pull in the volsync 7.1 deps (when we upgraded from older volsync). But when we pulled in newer velero code (post-1.11), the code we were referencing was making calls that were incompatible with the newer k8s libs, so we needed to revert to the version velero uses, which is possible now because we're removing the volsync dependency -- and the line above is one consequence of that change, reverting this change that was made at the same time as the lib change.

**Comment by sseago** [comment]
Looks like we're at the point now where I can debug individual e2e test failures. Will update PR periodically. Converting back to draft.

**Review by shubham-pampattiwar — APPROVED** [review]

**Review by kaovilai — APPROVED** [review]

**Comment by kaovilai** [comment]
If desired, tide query PR https://github.com/openshift/release/pull/41556

Tho it's not really necessary as it's not like there'll be a lot of PRs needing to be queued here. For this feature branch you can just hit the merge button.
